examples/fabric-sync/admin/DeviceManager.cpp

```diff
@@ -143,4 +216,264 @@ CHIP_ERROR DeviceManager::UnpairRemoteDevice(NodeId nodeId)
+
+        if (device == nullptr)
+        {
+            ChipLogProgress(NotSpecified, "No device on Endpoint: %u", endpoint);
+            continue;
+        }
+    }
+}
+
+void DeviceManager::SendCommissionNodeRequest(uint64_t requestId, uint16_t responseTimeoutSeconds)
+{
+    ChipLogProgress(NotSpecified, "Request the Commissioner Control Server to begin commissioning a previously approved request.");
+
+    CHIP_ERROR error = mCommissionerControl.CommissionNode(requestId, responseTimeoutSeconds);
+
+    if (error != CHIP_NO_ERROR)
+    {
+        ChipLogError(NotSpecified,
+                     "Failed to send CommissionNode command to the remote bridge (NodeId: %lu). Error: %" CHIP_ERROR_FORMAT,
+                     mRemoteBridgeNodeId, error.Format());
+        return;
+    }
+}
+
+void DeviceManager::HandleReverseOpenCommissioningWindow(TLV::TLVReader & data)
+{
+    ChipLogProgress(NotSpecified, "Handle ReverseOpenCommissioningWindow command.");
+
+    app::Clusters::CommissionerControl::Commands::ReverseOpenCommissioningWindow::DecodableType value;
+    CHIP_ERROR error = app::DataModel::Decode(data, value);
+
+    if (error != CHIP_NO_ERROR)
+    {
+        ChipLogError(NotSpecified, "Failed to decode command response value. Error: %" CHIP_ERROR_FORMAT, error.Format());
+        return;
+    }
+
+    // Log all fields
+    ChipLogProgress(NotSpecified, "DecodableType fields:");
+    ChipLogProgress(NotSpecified, "  commissioningTimeout: %u", value.commissioningTimeout);
+    ChipLogProgress(NotSpecified, "  discriminator: %u", value.discriminator);
+    ChipLogProgress(NotSpecified, "  iterations: %u", value.iterations);
+    ChipLogProgress(NotSpecified, "  PAKEPasscodeVerifier size: %lu", value.PAKEPasscodeVerifier.size());
+    ChipLogProgress(NotSpecified, "  salt size: %lu", value.salt.size());
+
+    OpenLocalBridgeCommissioningWindow(value.iterations, value.commissioningTimeout, value.discriminator,
+                                       ByteSpan(value.salt.data(), value.salt.size()),
+                                       ByteSpan(value.PAKEPasscodeVerifier.data(), value.PAKEPasscodeVerifier.size()));
+}
+
+void DeviceManager::HandleAttributeData(const app::ConcreteDataAttributePath & path, TLV::TLVReader & data)
+{
+    if (path.mClusterId == app::Clusters::Descriptor::Id &&
+        path.mAttributeId == app::Clusters::Descriptor::Attributes::PartsList::Id)
+    {
+        HandleAttributePartsListUpdate(data);
+        return;
+    }
+}
+
+void DeviceManager::HandleEventData(const app::EventHeader & header, TLV::TLVReader & data)
+{
+    if (header.mPath.mClusterId == app::Clusters::CommissionerControl::Id &&
+        header.mPath.mEventId == app::Clusters::CommissionerControl::Events::CommissioningRequestResult::Id)
+    {
+        HandleCommissioningRequestResult(data);
+    }
+}
+
+void DeviceManager::HandleCommandResponse(const app::ConcreteCommandPath & path, TLV::TLVReader & data)
```

Comment:
Should we be checking that the endpoint is what we expected. Assuming zap is configured there really shouldn't be a way to receive this on anything else I just think it is a decent sanity check

Reply:
This is to open the commissioning window of the local bridge, now, the local bridge is not part of local fabric in fabric-sync app, so, there is no associated endpoint to verify

Reply:
Sorry I selected the wrong line I meant to say checking the endpoint is aggregator endpoint before calling `HandleReverseOpenCommissioningWindow`

Reply:
Added